src/jukebox/jukebox_functions.py

```diff
@@ -9,134 +9,136 @@
-    scheduler = BlockingScheduler(timezone="CET")
-
-    # first songs starts after first round of voting (1 minute)
-    scheduler.add_job(play_next_song, 'date', run_date=first_round, args=[Session, scheduler, music_folder])
-    print('Starting Jukebox')
-    scheduler.start()
-
-    return scheduler
-
-
-def count_votes(session):
-    """
-
-    :param session: db session object
-    :return: object, database row with info of most voted song
-    """
-    now = datetime.now()
-    vote_round = session.query(Round).filter(and_(Round.start_date <= now, Round.end_date >= now)).first()
-
-    counted_votes = session.query(func.count(Votes.song_id), Votes, Songs). \
-        filter_by(round_id=vote_round.id). \
-        join(Songs). \
-        group_by(Votes.song_id).all()
-
-    # handle rounds without votes
-    if len(counted_votes) > 0:
-        winner = max(counted_votes, key=itemgetter(0))[2]  # Remove returned votes object instead of indexing to 2?
-    else:
-        random_song_id = session.query(SelectedSongs).filter(SelectedSongs.round_id == vote_round.id).first().song_id
-        winner = session.query(Songs).filter(Songs.id == random_song_id).first()
-
-    return winner
-
-
-def play_next_song(session_obj, scheduler, music_folder):
-    """
-    1. Call count votes
-    2. Setup New Round
-    3. Play next song
-
-    :param session_obj: db session object
-    :param scheduler: apschedular object
-    :param music_folder: path to music
-
-    :return: None
-    """
-    session = session_obj()
-    song = count_votes(session)
-    round_end = setup_new_round(session=session, song=song)
-
-    song_path = os.path.join(music_folder, song.filename)
-
-    run_date = round_end - timedelta(minutes=0, seconds=1)
-    print('New song at', run_date)
-    scheduler.add_job(play_next_song, 'date', run_date=run_date, args=[session_obj, scheduler, music_folder])
-
-    print('Playing:', song.filename)
-    # if on RasPi use 'vlc --one-instance --playlist-enqueue'
-    subprocess.call("afplay {}".format(song_path), shell=True)
-    # subprocess.call("vlc --one-instance --playlist-enqueue {}".format(song_path), shell=True)
-    return None
-
-
-def setup_new_round(session, first_round=False, song=None):
-    """
-    :param session: db session object
-    :param first_round: Default: False. True if this is the first round to setup in jukebox session,
-    :param song: song object to determine new round end
-    :return:
-    """
-    # Randomly provide selection of songs to choose from
-    songs = session.query(Songs).all()
-    selected_song_ids = random.sample(songs, 4)
-
-    if first_round:
-        print('Setting up initial round')
+class JukeBox:
+    def __init__(self, music_folder, db_path):
+        self.music_folder = music_folder
+        self.db_path = db_path
+        self.db = 'sqlite:///{}'.format(self.db_path)
+        self.engine = create_engine(self.db, echo=False)
+        self.Session = sessionmaker(bind=self.engine)
```

Comment:
You don't use the Session object other than for making the instance below (it's not passed to the scheduler in the init anymore), so there's no need to make it a class variable.